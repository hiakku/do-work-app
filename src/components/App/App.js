import { Box, CssBaseline } from "@mui/material";
import { useEffect, useState } from "react";
import Dashboard from "../Dashboard/Dashboard";
import Header from "../Header/Header";
import ModalContainer from "../Modal/Modal";
import { LocalizationProvider } from "@mui/x-date-pickers";
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";

function App() {
  const [openModal, setOpenModal] = useState(false);
  const [taskList, setTaskList] = useState([]);
  const [editableId, setEditableId] = useState("");
  useEffect(() => {
    const tasks = JSON.parse(localStorage["taskDetails"] || "[]");
    setTaskList(tasks);
  }, []);
  const handleOpenModal = () => {
    setOpenModal(true);
  };
  const handleCloseModal = () => {
    setOpenModal(false);
  };
  return (
    <LocalizationProvider dateAdapter={AdapterDayjs}>
      <Box sx={{ display: "flex", minHeight: "100vh" }}>
        <ModalContainer
          openModal={openModal}
          handleClose={handleCloseModal}
          setTaskList={setTaskList}
          taskList={taskList}
          setEditableId={setEditableId}
          editableId={editableId}
        />
        <CssBaseline />
        <Header handleOpenModal={handleOpenModal} />
        <Dashboard
          taskList={taskList}
          setTaskList={setTaskList}
          setEditableId={setEditableId}
          handleOpenModal={handleOpenModal}
        />
      </Box>
    </LocalizationProvider>
  );
}

export default App;
