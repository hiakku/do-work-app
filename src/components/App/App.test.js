import { render } from "@testing-library/react";
import App from "./App";

describe("Test for App component", () => {
  jest.mock("../Header/Header");
  it("Match snapshot for App component ", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
