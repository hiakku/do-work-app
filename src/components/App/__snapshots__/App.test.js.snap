// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for App component Match snapshot for App component  1`] = `
<div>
  <div
    class="MuiBox-root css-ogmp5q"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-pgnjvu-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-u4fbkj-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1svc8n9-MuiTypography-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="WorkIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
            />
          </svg>
          DoWork
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained  MuiButton-contained Primary MuiButton-sizeMedium MuiButton-contained SizeMedium MuiButton-root MuiButton-contained  MuiButton-contained Primary MuiButton-sizeMedium MuiButton-contained SizeMedium css-1or3yxj-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Create Task
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-vrs2x7-MuiContainer-root"
    >
      <div
        style="height: calc(100vh - 120px); width: 100%;"
      >
        <div
          class="MuiBox-root css-1v0qxk3"
        >
          <div>
            <h2
              class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
            >
              DoWork
            </h2>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-yu2ml3-MuiSvgIcon-root"
              data-testid="WorkIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
              />
            </svg>
            <br />
            Click on 
            <b>
              Create Task
            </b>
             and,
            <br />
             start creating new task to plan your day better!!
            <br />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for App component renders  1`] = `
<div>
  <div
    class="MuiBox-root css-ogmp5q"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-pgnjvu-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-u4fbkj-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1svc8n9-MuiTypography-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="WorkIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
            />
          </svg>
          DoWork
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained  MuiButton-contained Primary MuiButton-sizeMedium MuiButton-contained SizeMedium MuiButton-root MuiButton-contained  MuiButton-contained Primary MuiButton-sizeMedium MuiButton-contained SizeMedium css-1or3yxj-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Create Task
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-vrs2x7-MuiContainer-root"
    >
      <div
        style="height: calc(100vh - 120px); width: 100%;"
      >
        <div
          class="MuiBox-root css-1v0qxk3"
        >
          <div>
            <h2
              class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
            >
              DoWork
            </h2>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-yu2ml3-MuiSvgIcon-root"
              data-testid="WorkIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
              />
            </svg>
            <br />
            Click on 
            <b>
              Create Task
            </b>
             and,
            <br />
             start creating new task to plan your day better!!
            <br />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders  1`] = `
<div>
  <div
    class="MuiBox-root css-ogmp5q"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-pgnjvu-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-u4fbkj-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1svc8n9-MuiTypography-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="WorkIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
            />
          </svg>
          DoWork
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained  MuiButton-contained Primary MuiButton-sizeMedium MuiButton-contained SizeMedium MuiButton-root MuiButton-contained  MuiButton-contained Primary MuiButton-sizeMedium MuiButton-contained SizeMedium css-1or3yxj-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Create Task
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-vrs2x7-MuiContainer-root"
    >
      <div
        style="height: calc(100vh - 120px); width: 100%;"
      >
        <div
          class="MuiBox-root css-1v0qxk3"
        >
          <div>
            <h2
              class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
            >
              DoWork
            </h2>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-yu2ml3-MuiSvgIcon-root"
              data-testid="WorkIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
              />
            </svg>
            <br />
            Click on 
            <b>
              Create Task
            </b>
             and,
            <br />
             start creating new task to plan your day better!!
            <br />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
